### PR TITLE
Jbaker/generate docs refactor

### PR DESCRIPTION
**Pull request by jason-capsule42**
# Alaska Airlines Pull Request

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

## Summary:

This change updates the static doc partials and makes the template docs compile using the remote resources that now exist on the server.

## Type of change:

Please delete options that are not relevant.

- [x] New capability
- [x] Revision of an existing capability
- [ ] Infrastructure change (automation, etc.)
- [ ] Other (please elaborate)

## Checklist:

- [x] My update follows the CONTRIBUTING guidelines of this project
- [x] I have performed a self-review of my own update

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._

**Thank you for your submission!**<br>
-- Auro Design System Team

### CODE REVIEW

**Review by blackfalcon — CHANGES_REQUESTED** [review]
When running a build, the root `./README` file is never created. This is consistent with `npm test`, `npm run build:complete` for both labs and standard repo.

File: template/scripts/generateDocs.js [hunk 1]
```diff
@@ -1,14 +1,84 @@
...
+  outputDir: './demo'
 }
 
-const markdownPath = path.join(__dirname, '../**/*.md')
-markdownMagic(markdownPath, config, callback)
+const markdownPathDemo = path.join(__dirname, '../docs/partials/demo.md')
+
+markdownMagic(markdownPathDemo, configDemo, callbackDemo)
+
+
```
Comment (blackfalcon) [inline template/scripts/generateDocs.js:61]: remove line

File: template/scripts/generateDocs.js [hunk 2]
```diff
@@ -1,14 +1,84 @@
...
-const config = {
-  matchWord: 'AURO-GENERATED-CONTENT'
+const readmeTemplateUrl = 'https://raw.githubusercontent.com/AlaskaAirlines/WC-Generator/master/componentDocs/README.md'
+
+// Copy README.md template from static source
+const dir = './docTemplates';
+
+if (!fs.existsSync(dir)){
+  fs.mkdirSync(dir);
+}
```
Comment (blackfalcon) [inline template/scripts/generateDocs.js:14]:
```suggestion
if (!fs.existsSync(dir)){
  fs.mkdirSync(dir);
};
```

File: template/scripts/generateDocs.js [hunk 3]
```diff
@@ -1,14 +1,84 @@
...
+    }
+  });
+}
+
+const readmeTemplateFile = fs.createWriteStream(readmeFilePath);
+const request = https.get(readmeTemplateUrl, function(response) {
+  response.pipe(readmeTemplateFile);
+});
+
+// Compile `./docTemplates/README.md` -> `./README.md`
```
Comment (blackfalcon) [inline template/scripts/generateDocs.js:31]: Is this commented out code or a comment of the code? 

If a comment of the code, please use the `/** */ syntax.

File: template/scripts/generateDocs.js [hunk 4]
```diff
@@ -1,14 +1,84 @@
...
+const path = require('path');
 const chalk = require('chalk');
-const markdownMagic = require('markdown-magic')
+const markdownMagic = require('markdown-magic');
+const fs = require('fs');
+const https = require('https');
 
-const config = {
-  matchWord: 'AURO-GENERATED-CONTENT'
+const readmeTemplateUrl = 'https://raw.githubusercontent.com/AlaskaAirlines/WC-Generator/master/componentDocs/README.md'
```
[the remark this answers is not in the document]
Comment (blackfalcon) [reply template/scripts/generateDocs.js:7]: There are missing semi-colons, please address where needed.

**Review by blackfalcon — CHANGES_REQUESTED** [review]

**Comment by blackfalcon** [comment]
Thar be the issue! 

```
FILE NOT FOUND /Users/dalesande/src/auro/auro-test/docTemplates/partials/browserSupport.md
✔ demo/demo.md Updated
 Transforms run
  ⁕ FILE:src=./description.md 
  ⁕ FILE:src=./useCases.md 
  ⁕ FILE:src=./../../apiExamples/basic.html 
  ⁕ CODE:src=./../../apiExamples/basic.html 

Finished compiling demo documentation.
node:fs:585
  handleErrorFromBinding(ctx);
  ^

Error: ENOENT: no such file or directory, open '/Users/dalesande/src/auro/auro-test/docTemplates/partials/browserSupport.md'
    at Object.openSync (node:fs:585:3)
    at Object.readFileSync (node:fs:453:35)
    at Object.FILE (/Users/dalesande/src/auro/auro-test/node_modules/markdown-magic/lib/transforms/file.js:15:25)
    at updateContents (/Users/dalesande/src/auro/auro-test/node_modules/markdown-magic/lib/updateContents.js:24:56)
    at processFile (/Users/dalesande/src/auro/auro-test/node_modules/markdown-magic/lib/processFile.js:101:32)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async /Users/dalesande/src/auro/auro-test/node_modules/markdown-magic/index.js:33:15
    at async Promise.all (index 0)
    at async markdownMagic (/Users/dalesande/src/auro/auro-test/node_modules/markdown-magic/index.js:37:18) {
  errno: -2,
  syscall: 'open',
  code: 'ENOENT',
  path: '/Users/dalesande/src/auro/auro-test/docTemplates/partials/browserSupport.md'
}
./auro-test (main *) $ 
```

**Review by blackfalcon — APPROVED** [review]

**Comment by jason-capsule42** [comment]
> Thar be the issue!
> 
> ```
> FILE NOT FOUND /Users/dalesande/src/auro/auro-test/docTemplates/partials/browserSupport.md
> ✔ demo/demo.md Updated
>  Transforms run
>   ⁕ FILE:src=./description.md 
>   ⁕ FILE:src=./useCases.md 
>   ⁕ FILE:src=./../../apiExamples/basic.html 
>   ⁕ CODE:src=./../../apiExamples/basic.html 
> 
> Finished compiling demo documentation.
> node:fs:585
>   handleErrorFromBinding(ctx);
>   ^
> 
> Error: ENOENT: no such file or directory, open '/Users/dalesande/src/auro/auro-test/docTemplates/partials/browserSupport.md'
>     at Object.openSync (node:fs:585:3)
>     at Object.readFileSync (node:fs:453:35)
>     at Object.FILE (/Users/dalesande/src/auro/auro-test/node_modules/markdown-magic/lib/transforms/file.js:15:25)
>     at updateContents (/Users/dalesande/src/auro/auro-test/node_modules/markdown-magic/lib/updateContents.js:24:56)
>     at processFile (/Users/dalesande/src/auro/auro-test/node_modules/markdown-magic/lib/processFile.js:101:32)
>     at processTicksAndRejections (node:internal/process/task_queues:96:5)
>     at async /Users/dalesande/src/auro/auro-test/node_modules/markdown-magic/index.js:33:15
>     at async Promise.all (index 0)
>     at async markdownMagic (/Users/dalesande/src/auro/auro-test/node_modules/markdown-magic/index.js:37:18) {
>   errno: -2,
>   syscall: 'open',
>   code: 'ENOENT',
>   path: '/Users/dalesande/src/auro/auro-test/docTemplates/partials/browserSupport.md'
> }
> ./auro-test (main *) $ 
> ```

Issue here is that the the generateDpcs script imports the readme.md template from the server before compiling it. The current version on the server doesnt use the static resources it wants to use local resources. As soon as this is actually on the master branch it will work correctly.

**Pull request merged**